AsyncDisplayKit/ASMapNode.mm

```diff
@@ -17,70 +17,80 @@ @interface ASMapNode()
   MKMapSnapshotter *_snapshotter;
   MKMapSnapshotOptions *_options;
   NSArray *_annotations;
-  ASDisplayNode *_mapNode;
   CLLocationCoordinate2D _centerCoordinateOfMap;
 }
 @end
 
 @implementation ASMapNode
 
-@synthesize liveMap = _liveMap;
 @synthesize needsMapReloadOnBoundsChange = _needsMapReloadOnBoundsChange;
 @synthesize mapDelegate = _mapDelegate;
+@synthesize region = _region;
 
-- (instancetype)initWithCoordinate:(CLLocationCoordinate2D)coordinate
+#pragma mark - Lifecycle
+- (instancetype)init
 {
   if (!(self = [super init])) {
     return nil;
   }
   self.backgroundColor = ASDisplayNodeDefaultPlaceholderColor();
   self.clipsToBounds = YES;
-
+  
   _needsMapReloadOnBoundsChange = YES;
-  _liveMap = NO;
+  
   _centerCoordinateOfMap = kCLLocationCoordinate2DInvalid;
-
+  _region = MKCoordinateRegionMake(CLLocationCoordinate2DMake(43.432858, 13.183671), MKCoordinateSpanMake(0.2, 0.2));
+  
   _options = [[MKMapSnapshotOptions alloc] init];
-  _options.region = MKCoordinateRegionMakeWithDistance(coordinate, 1000, 1000);;
+  _options.region = _region;
   
   return self;
 }
 
-- (void)setAnnotations:(NSArray *)annotations
+- (void)didLoad
 {
-  ASDN::MutexLocker l(_propertyLock);
-  _annotations = [annotations copy];
-  if (annotations.count != _annotations.count) {
-    // Redraw
-    [self setNeedsDisplay];
+  [super didLoad];
+  if ([self wasLiveMapPreviously]) {
+    self.userInteractionEnabled = YES;
+    [self addLiveMap];
   }
 }
 
-- (void)setUpSnapshotter
+- (void)fetchData
 {
-  if (!_snapshotter) {
-    ASDisplayNodeAssert(!CGSizeEqualToSize(CGSizeZero, self.calculatedSize), @"self.calculatedSize can not be zero. Make sure that you are setting a preferredFrameSize or wrapping ASMapNode in a ASRatioLayoutSpec or similar.");
-      _options.size = self.calculatedSize;
-      _snapshotter = [[MKMapSnapshotter alloc] initWithOptions:_options];
+  [super fetchData];
+  if ([self wasLiveMapPreviously]) {
+    [self addLiveMap];
+  } else {
+    [self setUpSnapshotter];
+    [self takeSnapshot];
   }
 }
 
+- (void)clearFetchedData
+{
+  [super clearFetchedData];
+  if (self.isLiveMap) {
+    [self removeLiveMap];
+  }
+}
+
+#pragma mark - Settings
+
 - (BOOL)isLiveMap
 {
-  ASDN::MutexLocker l(_propertyLock);
-  return _liveMap;
+  return (_mapView != nil);
 }
 
 - (void)setLiveMap:(BOOL)liveMap
 {
-  ASDN::MutexLocker l(_propertyLock);
-  if (liveMap == _liveMap) {
-    return;
-  }
-  _liveMap = liveMap;
   liveMap ? [self addLiveMap] : [self removeLiveMap];
 }
 
+- (BOOL)wasLiveMapPreviously
```

Comment:
Hmm, the need for this might challenge my original thought of eliminating the liveMap BOOL...I think we may need it after all.  Most particularly, what happens if you allocate the map node (on a background thread as nodes are supposed to be initialized) and then immediately setLiveMap:YES, because you want your map node to always be live once it comes onscreen?  We shouldn't allocate the node then - it should be done AT EARLIEST in -fetchData - but this would require the prior implementation that you had with the _propertyLocked BOOL.